internal/bffauth/client.go

```diff
@@ -0,0 +1,202 @@
+package bffauth
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/bluesky-social/indigo/atproto/crypto"
+)
+
+type ExternalHabitatUser struct {
+	DID       string
+	PublicKey crypto.PublicKey
+	Host      string
+}
+
+func getTempStores() (map[string]crypto.PrivateKey, map[string]*ExternalHabitatUser, error) {
+	testKey, err := crypto.GeneratePrivateKeyP256()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	testPubKey, err := testKey.PublicKey()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	testUser := &ExternalHabitatUser{
+		DID:       "did:plc:avkzrsj7u4kmoq33gx6v4lh4",
+		PublicKey: testPubKey,
+		Host:      "beacon-dev.tail07d32.ts.net",
+	}
+
+	return map[string]crypto.PrivateKey{
+			"did:plc:avkzrsj7u4kmoq33gx6v4lh4": testKey,
+		}, map[string]*ExternalHabitatUser{
+			"did:plc:avkzrsj7u4kmoq33gx6v4lh4": testUser,
+		}, nil
+
+}
+
+type Client struct {
+	// This is the store of temporary friends that are added to the BFF.
```

Comment:
this should actually be an `interface` (see https://github.com/eagraf/habitat-new/pull/55/files#diff-3fcf2c4d8538ad0e112b07632d17fd396d921e24861c08a811370b57277ad8abR11-R13) so it can be mocked out